cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -531,18 +540,75 @@ angular.module(PKG.name + '.commons')
       });
     }
 
+    function getPosition(e) {
+      var posx = 0;
+      var posy = 0;
+
+      if (e.pageX || e.pageY) {
+        posx = e.pageX;
+        posy = e.pageY;
+      } else if (e.clientX || e.clientY) {
+        posx = e.clientX + document.body.scrollLeft + document.documentElement.scrollLeft;
+        posy = e.clientY + document.body.scrollTop + document.documentElement.scrollTop;
+      }
+
+      return {
```

Comment:
We might want to rethink the contextmenu here. I was a little unclear as to how the context menu worked and then realized that on connections select we show a context menu but otherwise we don't. Also the context menu is shown near the cursor (which is correct) but I don't think would want a context menu on the entire dag. Might be on the connections alone for now? The reason I am saying this is because of this,


<img width="1680" alt="screen shot 2017-08-09 at 8 23 36 am" src="https://user-images.githubusercontent.com/1452845/29129684-774ad4b8-7cdc-11e7-9e89-7583619db18b.png">

I have selected the connections and right clicked elsewhere in the canvas. So I am seeing a context menu for the canvas but the delete option is for the connections I have selected.